src/ufoLib2/objects/lib.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+from base64 import b85decode, b85encode
+from typing import TYPE_CHECKING, Any, Callable, ClassVar, Dict, Mapping, Tuple, Union
+
+import attr
+from attr import field, frozen
+
+from ufoLib2.constants import DATA_LIB_KEY
+
+if TYPE_CHECKING:
+    from typing import Type
+
+    from cattr import GenConverter
+
+# unfortunately mypy is not smart enough to support recursive types like plist...
+# PlistEncodable = Union[
+#     bool,
+#     bytes,
+#     datetime,
+#     float,
+#     int,
+#     str,
+#     Mapping[str, PlistEncodable],
+#     Sequence[PlistEncodable],
+# ]
+
+
+DEFAULT_ENCODING = "base85"
+
+BYTES_ENCODINGS: dict[
```

Comment:
```suggestion
BYTES_ENCODINGS: Dict[
```

Reply:
I tried to use the fancy generic types for the postponed annotations that are only used for static typechecking (like in this line), and only use the old-style generic collections with uppercase letters for the places where we need those types at runtime (for cattrs structuring/destructuring or when are used as base class).